src/dataEntryApp/views/subjectDashBoard/components/subjectDashboardProgramTab.js

```diff
@@ -0,0 +1,133 @@
+
+  return <Typography>{value == index && <Box p={3}>{children}</Box>}</Typography>;
+}
+
+TabPanel.propTypes = {
+  children: PropTypes.node,
+  index: PropTypes.any.isRequired,
+  value: PropTypes.any.isRequired
+};
+
+let flagActive = true;
+
+const SubjectDashboardProgramTab = ({ program }) => {
+  const [value, setValue] = React.useState(0);
+  const [value1, setValue1] = React.useState(0);
+
+  const handleChangeActive = (event, newValue) => {
+    flagActive = true;
+    setValue(newValue);
+  };
+
+  const handleChangeExited = (event, newvalue) => {
+    setValue1(newvalue);
+    flagActive = false;
+  };
+
+  const classes = useStyles();
+
+  return (
+    <Fragment>
+      <Paper className={classes.root}>
+        <Grid className={classes.tabView} container spacing={1}>
+          <Grid xs={6}>
+            <label>Active Programs</label>
+            <AppBar position="static" color="default">
+              <Tabs
+                onChange={handleChangeActive}
+                value={value}
+                indicatorColor="primary"
+                textColor="primary"
+                variant="scrollable"
+                scrollButtons="auto"
+                aria-label="scrollable auto tabs example"
+              >
+                {program && program.enrolments
+                  ? program.enrolments.map((element, index) =>
+                      element.programExitDateTime == null ? (
+                        <Tab label={element.operationalProgramName} />
+                      ) : (
+                        ""
+                      )
+                    )
+                  : ""}
+              </Tabs>
+            </AppBar>
+          </Grid>
+
+          <Grid xs={2} />
+          <Grid xs={4}>
+            <label>Exited Programs</label>
+            <AppBar position="static" color="default">
+              <Tabs
+                value={value1}
+                onChange={handleChangeExited}
+                indicatorColor="primary"
+                textColor="primary"
+                variant="scrollable"
+                scrollButtons="auto"
+                aria-label="scrollable auto tabs example"
+              >
+                {program && program.enrolments
+                  ? program.enrolments.map((element, index) =>
+                      element.programExitDateTime != null ? (
+                        <Tab label={element.operationalProgramName} />
+                      ) : (
+                        ""
+                      )
+                    )
+                  : ""}
+              </Tabs>
+            </AppBar>
+          </Grid>
+        </Grid>
+        {flagActive && program && program.enrolments
+          ? program.enrolments.map((element, index) => (
+              <Fragment>
+                <TabPanel value={value} index={index}>
+                  <ProgramView programData={element} />
+                </TabPanel>
+              </Fragment>
+            ))
+          : ""}
```

Comment:
avoid the duplication here.

Reply:
@RashmiDavandePsl This is still not resolved. You are duplicating pretty much the same code i.e. code between line 110 to line 118 is duplicate of 119 to 127.